Add files via upload

Artem22831 opened the pull request.

Artem22831 committed:
Add files via upload

Artem22831 committed:
Add files via upload

Nastasia8 commented on Untitled-1.txt:
```diff
@@ -0,0 +1,55 @@
+    if ql <= l and qr >= r:
+        return do[v]
+    if ql >= r or qr <= l:
+        return -1e9
```
Мы с Вами забыли поправить код 22
```
from math import gcd
def build(v, l, r, do, a):
    if r-l == 1:
        do[v] = a[l]
        return
    m = (l+r)//2
    build(2 * v + 1, l, m, do, a)
    build(2 * v + 2, m, r, do, a)
    do[v] = gcd(do[2 * v + 1], do[2 * v + 2])
def get_gcd(v, l, r, do, ql, qr):
    if ql <= l and qr >= r:
        return do[v]
    if ql >= r or qr <= l:
        return 0
    m = (l+r)//2
    tl = get_gcd(2 * v + 1, l, m, do, ql, qr)
    tr = get_gcd(2 * v + 2, m, r, do, ql, qr)
    return gcd(tl, tr)
def update(v, l, r, do, idx, value):
    if r-l == 1:
        do[v] = value
        return
    m = (l+r)//2
    if idx < m:
        update(v * 2 + 1, l, m, do, idx, value)
    else:
        update(v * 2 + 2, m, r, do, idx, value)
    do[v] = gcd(do[v * 2 + 1], do[v * 2 + 2])

n = int(input())
do = [0] * (4 * n)
a = list(map(int, input().split()))[:n]
build(0, 0, n, do, a)
q = int(input())
res = []
while q !=0:
    que, l, r = map(str, input().split())
    if que == "s":
        res.append(get_gcd(0, 0, n, do, int(l) - 1, int(r)))
    else:
        update(0, 0, n, do, int(l) - 1, int(r))
    q-=1
print(*res)
```

Nastasia8 replied:
Код рабочий, я его проверяла, и в IDE точно работает. 
Если Вы используете какой-нибудь онлайн-компилятор, то просто замените его на другой, например,  https://www.online-python.com/ 
Кстати, я Вас не нашла на Вкладке "Положение участников" в 5 модуле, не вижу чтобы Вы тестировали код.

Artem22831 committed:
Add files via upload

Artem22831 commented:
Здравствуйте , у меня почему то не работает код на яндекс контексте и в
компиляторе кстати тоже .

пн, 26 апр. 2021 г. в 16:09, Nastasia8 ***@***.***>:

> *@Nastasia8* commented on this pull request.
> ------------------------------
>
> In Untitled-1.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r620277941>
> :
>
> > +    return (a + b)
>
> +
>
> +def build(v, l , r, do, a):
>
> +    if r-l == 1:
>
> +        do[v] = a[l]
>
> +        return
>
> +    m = (l+r)//2
>
> +    build(2*v + 1, l, m, do, a)
>
> +    build(2*v + 2, m, r, do, a)
>
> +    do[v] = gcd(do[2*v + 1], do[2*v + 2])
>
> +
>
> +def get_max(v, l , r, do , ql, qr):
>
> +    if ql <= l and qr >= r:
>
> +        return do[v]
>
> +    if ql >= r or qr <= l:
>
> +        return -1e9
>
>
> Мы с Вами забыли поправить код 22
>
> from math import gcd
>
> def build(v, l, r, do, a):
>
>     if r-l == 1:
>
>         do[v] = a[l]
>
>         return
>
>     m = (l+r)//2
>
>     build(2 * v + 1, l, m, do, a)
>
>     build(2 * v + 2, m, r, do, a)
>
>     do[v] = gcd(do[2 * v + 1], do[2 * v + 2])
>
> def get_gcd(v, l, r, do, ql, qr):
>
>     if ql <= l and qr >= r:
>
>         return do[v]
>
>     if ql >= r or qr <= l:
>
>         return 0
>
>     m = (l+r)//2
>
>     tl = get_gcd(2 * v + 1, l, m, do, ql, qr)
>
>     tr = get_gcd(2 * v + 2, m, r, do, ql, qr)
>
>     return gcd(tl, tr)
>
> def update(v, l, r, do, idx, value):
>
>     if r-l == 1:
>
>         do[v] = value
>
>         return
>
>     m = (l+r)//2
>
>     if idx < m:
>
>         update(v * 2 + 1, l, m, do, idx, value)
>
>     else:
>
>         update(v * 2 + 2, m, r, do, idx, value)
>
>     do[v] = gcd(do[v * 2 + 1], do[v * 2 + 2])
>
>
>
> n = int(input())
>
> do = [0] * (4 * n)
>
> a = list(map(int, input().split()))[:n]
>
> build(0, 0, n, do, a)
>
> q = int(input())
>
> res = []
>
> while q !=0:
>
>     que, l, r = map(str, input().split())
>
>     if que == "s":
>
>         res.append(get_gcd(0, 0, n, do, int(l) - 1, int(r)))
>
>     else:
>
>         update(0, 0, n, do, int(l) - 1, int(r))
>
>     q-=1
>
> print(*res)
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#pullrequestreview-644648112>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTPONUA7B6GZAJFKXPTTKVQZJANCNFSM43RRTIWA>
> .
>

Nastasia8 commented on 5 modul/3.txt:
```diff
@@ -0,0 +1,36 @@
+    build(2*v+2, m, r, do, a)
+    do[v] = global(do[2*v+1], do[2*v+2])
+
+def global(v, l, r, do, ql, qr):
```
Что значит global???
Функции должны иметь осмысленные названия.

Artem22831 commented:
нет я тестировал, вот

ср, 5 мая 2021 г. в 20:28, Nastasia8 ***@***.***>:

> *@Nastasia8* commented on this pull request.
> ------------------------------
>
> In 5 modul/3.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r626762383>
> :
>
> > @@ -0,0 +1,36 @@
>
> +from math import gcd
>
> +
>
> +def build(v, l, r, do, a):
>
> +    if r-l == 1:
>
> +        do[v] = a[l]
>
> +        return
>
> +    m = (r+l)//2
>
> +    build(2*v+1, l, m, do, a)
>
> +    build(2*v+2, m, r, do, a)
>
> +    do[v] = global(do[2*v+1], do[2*v+2])
>
> +
>
> +def global(v, l, r, do, ql, qr):
>
>
> Что значит global???
> Функции должны иметь осмысленные названия.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#pullrequestreview-652570191>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTPIAM6G53X3XWOQ5ETTMF55HANCNFSM43RRTIWA>
> .
>

Artem22831 committed:
Add files via upload

Nastasia8 commented on 4 modul/3.txt:
```diff
@@ -0,0 +1,10 @@
+        stack.pop()
+    index[i] = stack[-1] if stack else -1
+    stack.append(i)
+print(*index)
```
ок.

Nastasia8 commented:
Так и не вижу Вас в 5 модуле на Контесте на вкладке Положение участников.
Скиньте мне скрин этой вкладки, хочу посмотреть на Ваши попытки (т.е. как Вы тестировали задачи).

Artem22831 commented:
нет, вот я отправлял  задания , я записан по эл. почте

пн, 10 мая 2021 г. в 13:07, Nastasia8 ***@***.***>:

> Так и не вижу Вас в 5 модуле на Контесте на вкладке Положение участников.
> Скиньте мне скрин этой вкладки, хочу посмотреть на Ваши попытки (т.е. как
> Вы тестировали задачи).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-836504499>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTIIS7JIJNHHIUG7GMLTM6V4TANCNFSM43RRTIWA>
> .
>

Nastasia8 commented:
Я знаю, что Вы записаны по почте -> artembakiev77@gmail.com, но на вкладке "Положение участников" Вас нет.
![11](https://user-images.githubusercontent.com/40060674/117648899-7bcad300-b197-11eb-84d6-592ee5589c32.jpg)
И через поиск Ваши результаты не находит
![11](https://user-images.githubusercontent.com/40060674/117649236-f398fd80-b197-11eb-88d3-b4f49c72dd91.jpg)
Вот:
![11](https://user-images.githubusercontent.com/40060674/117649355-162b1680-b198-11eb-9245-3e6c37548e48.jpg)

Artem22831 commented:
я незнаю почему так

пн, 10 мая 2021 г. в 14:01, Nastasia8 ***@***.***>:

> Я знаю, что Вы записаны по почте -> ***@***.***, но на
> вкладке "Положение участников" Вас нет.
> [image: 11]
> <https://user-images.githubusercontent.com/40060674/117648899-7bcad300-b197-11eb-84d6-592ee5589c32.jpg>
> И через поиск Ваши результаты не находит
> [image: 11]
> <https://user-images.githubusercontent.com/40060674/117649236-f398fd80-b197-11eb-88d3-b4f49c72dd91.jpg>
> Вот:
> [image: 11]
> <https://user-images.githubusercontent.com/40060674/117649355-162b1680-b198-11eb-9245-3e6c37548e48.jpg>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-836555204>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTO4DCKTSHZVI6EWRCLTM64ITANCNFSM43RRTIWA>
> .
>

Artem22831 committed:
Add files via upload

Nastasia8 commented on 6 modul/1.txt:
```diff
@@ -0,0 +1,38 @@
+    list_ = list(map(int, input().split()))
+    heap = build_heap(list_)
+    print(*heap)
+main()  
```
def buid_bink(arr) или build_heap(list_) (функция одна и та же, но названия разные)
Потом, в функции shift_down() -> откуда взялся index, если там только current_index???

```
def build_heap(arr):
    heap = arr[:]
    for i in range(len(heap)-1, -1, -1):
        shift_down(i, heap)
    return heap

def shift_up(current_index, heap):
    while heap[current_index] < heap[(current_index-1)//2]:
        heap[current_index], heap[len(current_index-1)//2] = heap[(current_index-1)//2], heap[current_index]
        current_index = (current_index-1)//2

def shift_down(current_index, heap):
    while 2*current_index+1 < len(heap):
        left_index = 2*current_index+1
        right_index = 2*current_index+2
        child_index = left_index
        if right_index < len(heap) and heap[left_index] > heap[right_index]:
            child_index = right_index
        if heap[child_index] >= heap[current_index]:
            break
        heap[current_index], heap[child_index] = heap[child_index], heap[current_index]
        current_index = child_index

def add(component, heap):
    heap.append(component)
    shift_up(len(heap)-1, heap)

def extract(heap):
    heap[0], heap[len(heap)-1] = heap[len(heap)-1], heap[0]
    heap.pop()
    shift_down(0, heap)

def min(heap):
    return heap[0]

def main():
    n = int(input())
    list_ = list(map(int, input().split()))
    heap = build_heap(list_)
    print(*heap)
    while len(heap):
        print(min(heap), end=" ")
        extract(heap)
main()
```
Куча построена по минимуму. Код рабочий.  Используете для его для решения задач на контесте.

Artem22831 commented:
Спасибо)

ср, 19 мая 2021 г., 18:40 Nastasia8 ***@***.***>:

> *@Nastasia8* commented on this pull request.
> ------------------------------
>
> In 6 modul/1.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r635363630>
> :
>
> > +
>
> +def add(component, heap):
>
> +    heap.append(component)
>
> +    shift_up(len(heap)-1, heap)
>
> +
>
> +def extract(heap):
>
> +    heap[0], heap[len(heap)-1] = heap[len(heap)-1], heap[0]
>
> +    heap.pop()
>
> +    shift_down(0, heap)
>
> +
>
> +def main():
>
> +    n = int(input())
>
> +    list_ = list(map(int, input().split()))
>
> +    heap = build_heap(list_)
>
> +    print(*heap)
>
> +main()
>
>
> def buid_bink(arr) или build_heap(list_) (функция одна и та же, но
> названия разные)
> Потом, в функции shift_down() -> откуда взялся index, если там только
> current_index???
>
> def build_heap(arr):
>
>     heap = arr[:]
>
>     for i in range(len(heap)-1, -1, -1):
>
>         shift_down(i, heap)
>
>     return heap
>
>
>
> def shift_up(current_index, heap):
>
>     while heap[current_index] < heap[(current_index-1)//2]:
>
>         heap[current_index], heap[len(current_index-1)//2] = heap[(current_index-1)//2], heap[current_index]
>
>         current_index = (current_index-1)//2
>
>
>
> def shift_down(current_index, heap):
>
>     while 2*current_index+1 < len(heap):
>
>         left_index = 2*current_index+1
>
>         right_index = 2*current_index+2
>
>         child_index = left_index
>
>         if right_index < len(heap) and heap[left_index] > heap[right_index]:
>
>             child_index = right_index
>
>         if heap[child_index] >= heap[current_index]:
>
>             break
>
>         heap[current_index], heap[child_index] = heap[child_index], heap[current_index]
>
>         current_index = child_index
>
>
>
> def add(component, heap):
>
>     heap.append(component)
>
>     shift_up(len(heap)-1, heap)
>
>
>
> def extract(heap):
>
>     heap[0], heap[len(heap)-1] = heap[len(heap)-1], heap[0]
>
>     heap.pop()
>
>     shift_down(0, heap)
>
>
>
> def min(heap):
>
>     return heap[0]
>
>
>
> def main():
>
>     n = int(input())
>
>     list_ = list(map(int, input().split()))
>
>     heap = build_heap(list_)
>
>     print(*heap)
>
>     while len(heap):
>
>         print(min(heap), end=" ")
>
>         extract(heap)
>
> main()
>
>
> Куча построена по минимуму. Код рабочий. Используете для его для решения
> задач на контесте.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#pullrequestreview-663372096>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTODHDS43A57TERLS5TTOPLWXANCNFSM43RRTIWA>
> .
>

Artem22831 committed:
Update 1.txt

Artem22831 committed:
Add files via upload

Artem22831 committed:
Add files via upload

Nastasia8 commented on 6 modul/3.txt:
```diff
@@ -0,0 +1,48 @@
+            index = child_index
+
+    def extract(self): #функция для извлечения 
+        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
```
Тут происходит обмен значений, первый элемент меняется с последним элементом в куче.

Nastasia8 commented on 6 modul/3.txt:
```diff
@@ -0,0 +1,48 @@
+
+    def extract(self): #функция для извлечения 
+        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
+        value=self.__heap.pop() #потом извлекается сюда
```
Происходит удаление последнего элемента из кучи и сохранение его в переменную value

Nastasia8 replied:
Нужно как-то так комментировать.

Nastasia8 commented on 5 modul/5.txt:
```diff
@@ -0,0 +1,48 @@
+            update(0, 0, n, do, int(l)-1, int(r))
+        q -= 1
+    print(*res)
+main()
```
У Вас в 5 модуле на Контесте зачтена только 2 задача, может быть мне поступить также?

Nastasia8 commented on 6 modul/1.txt:
```diff
@@ -0,0 +1,62 @@
+    else:
+        return 0, id
+
+def min(heap): #поиск минимального
```
Извлечение вершины кучи, построенной по минимальному элементу.

Nastasia8 commented on 6 modul/1.txt:
```diff
@@ -0,0 +1,62 @@
+            print(*i)
+        else:
+            print(i)
+    print(*heap)
```
Ну, тут явно нужны комментарии для функции main()
Не забываем условие: если копируем код, то он должен быть максимально закомментирован.
Это касается всего модуля 6.

Nastasia8 replied:
И кстати, Вы эти задачи даже на Контест не загрузили.

Artem22831 commented:
У меня на контесте в 6 модуле прошла только 3 задача ,которую я
копипастнул , а 1 и 2 задача не прошла я их сам делал, в 5 модуле 3 и 5
задача не прошла тоже , только 2 прошла

вт, 25 мая 2021 г. в 17:43, Nastasia8 ***@***.***>:

> *@Nastasia8* commented on this pull request.
> ------------------------------
>
> In 6 modul/3.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638850751>
> :
>
> > +
> +
> +    def shift_down(self, index):
> +        while 2*index+1 < self.length:
> +            left_index = 2*index+1
> +            right_index = 2*index+2
> +            child_index = left_index
> +            if right_index < self.length and self.__heap[left_index] < self.__heap[right_index]:
> +                child_index = right_index #child_index
> +            if self.__heap[child_index] <= self.__heap[index]:
> +                break
> +            self.__heap[index], self.__heap[child_index] = self.__heap[child_index], self.__heap[index]
> +            index = child_index
> +
> +    def extract(self): #функция для извлечения
> +        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
>
> Тут происходит обмен значений, первый элемент меняется с последним
> элементом в куче.
> ------------------------------
>
> In 6 modul/3.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638851600>
> :
>
> > +
> +    def shift_down(self, index):
> +        while 2*index+1 < self.length:
> +            left_index = 2*index+1
> +            right_index = 2*index+2
> +            child_index = left_index
> +            if right_index < self.length and self.__heap[left_index] < self.__heap[right_index]:
> +                child_index = right_index #child_index
> +            if self.__heap[child_index] <= self.__heap[index]:
> +                break
> +            self.__heap[index], self.__heap[child_index] = self.__heap[child_index], self.__heap[index]
> +            index = child_index
> +
> +    def extract(self): #функция для извлечения
> +        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
> +        value=self.__heap.pop() #потом извлекается сюда
>
> Происходит удаление последнего элемента из кучи и сохранение его в
> переменную value
> ------------------------------
>
> In 5 modul/5.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638852776>
> :
>
> > +def main():
> +    n = int(input())
> +    do = [0]*4*n
> +    a = list(map(int, input().split()))[:n]
> +    build(0, 0, n, do, a)
> +    q = int(input())
> +    res = []
> +    while q != 0:
> +        que, l, r = map(str, input().split())
> +        if que=="s":
> +            res.append(get_gcd(0, 0, n, do, int(l)-1, int(r)))
> +        else:
> +            update(0, 0, n, do, int(l)-1, int(r))
> +        q -= 1
> +    print(*res)
> +main()
>
> У Вас в 5 модуле на Контесте зачтена только 2 задача, может быть мне
> поступить также?
> ------------------------------
>
> In 6 modul/1.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638853989>
> :
>
> > +    return child_index+1
> +
> +def add(component, heap):
> +    heap.append(component)
> +    return shift_up(len(heap)-1, heap)
> +
> +
> +def extract(heap):
> +    heap[0], heap[len(heap)-1] = heap[len(heap)-1], heap[0]
> +    id = heap.pop()
> +    if heap:
> +        return shift_down(0, heap), id
> +    else:
> +        return 0, id
> +
> +def min(heap): #поиск минимального
>
> Извлечение вершины кучи, построенной по минимальному элементу.
> ------------------------------
>
> In 6 modul/1.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638855902>
> :
>
> > +        if int(data[0]) == 1:
> +            if not heap:
> +                result.append(-1)
> +            else:
> +                result.append(extract(heap))
> +        elif int(data[0]) == 2:
> +            if len(heap) == n:
> +                result.append(-1)
> +            else:
> +                result.append(add(int(data[-1]), heap))
> +    for i in rasult:
> +        if type(i) == tuple:
> +            print(*i)
> +        else:
> +            print(i)
> +    print(*heap)
>
> Ну, тут явно нужны комментарии для функции main()
> Не забываем условие: если копируем код, то он должен быть максимально
> закомментирован.
> Это касается всего модуля 6.
> ------------------------------
>
> In 6 modul/3.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638851790>
> :
>
> > +
> +    def shift_down(self, index):
> +        while 2*index+1 < self.length:
> +            left_index = 2*index+1
> +            right_index = 2*index+2
> +            child_index = left_index
> +            if right_index < self.length and self.__heap[left_index] < self.__heap[right_index]:
> +                child_index = right_index #child_index
> +            if self.__heap[child_index] <= self.__heap[index]:
> +                break
> +            self.__heap[index], self.__heap[child_index] = self.__heap[child_index], self.__heap[index]
> +            index = child_index
> +
> +    def extract(self): #функция для извлечения
> +        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
> +        value=self.__heap.pop() #потом извлекается сюда
>
> Нужно как-то так комментировать.
> ------------------------------
>
> In 6 modul/1.txt
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638856161>
> :
>
> > +        if int(data[0]) == 1:
> +            if not heap:
> +                result.append(-1)
> +            else:
> +                result.append(extract(heap))
> +        elif int(data[0]) == 2:
> +            if len(heap) == n:
> +                result.append(-1)
> +            else:
> +                result.append(add(int(data[-1]), heap))
> +    for i in rasult:
> +        if type(i) == tuple:
> +            print(*i)
> +        else:
> +            print(i)
> +    print(*heap)
>
> И кстати, Вы эти задачи даже на Контест не загрузили.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#pullrequestreview-667914222>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTOJH3EJFO7KRNJGZJ3TPOZSFANCNFSM43RRTIWA>
> .
>

Artem22831 commented:
Я вам 3 и 5 задачу из 5 модуля скидывал на гит ещё давно

вт, 25 мая 2021 г. в 17:53, Jumper csgo ***@***.***>:

> У меня на контесте в 6 модуле прошла только 3 задача ,которую я
> копипастнул , а 1 и 2 задача не прошла я их сам делал, в 5 модуле 3 и 5
> задача не прошла тоже , только 2 прошла
>
> вт, 25 мая 2021 г. в 17:43, Nastasia8 ***@***.***>:
>
>> *@Nastasia8* commented on this pull request.
>> ------------------------------
>>
>> In 6 modul/3.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638850751>
>> :
>>
>> > +
>> +
>> +    def shift_down(self, index):
>> +        while 2*index+1 < self.length:
>> +            left_index = 2*index+1
>> +            right_index = 2*index+2
>> +            child_index = left_index
>> +            if right_index < self.length and self.__heap[left_index] < self.__heap[right_index]:
>> +                child_index = right_index #child_index
>> +            if self.__heap[child_index] <= self.__heap[index]:
>> +                break
>> +            self.__heap[index], self.__heap[child_index] = self.__heap[child_index], self.__heap[index]
>> +            index = child_index
>> +
>> +    def extract(self): #функция для извлечения
>> +        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
>>
>> Тут происходит обмен значений, первый элемент меняется с последним
>> элементом в куче.
>> ------------------------------
>>
>> In 6 modul/3.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638851600>
>> :
>>
>> > +
>> +    def shift_down(self, index):
>> +        while 2*index+1 < self.length:
>> +            left_index = 2*index+1
>> +            right_index = 2*index+2
>> +            child_index = left_index
>> +            if right_index < self.length and self.__heap[left_index] < self.__heap[right_index]:
>> +                child_index = right_index #child_index
>> +            if self.__heap[child_index] <= self.__heap[index]:
>> +                break
>> +            self.__heap[index], self.__heap[child_index] = self.__heap[child_index], self.__heap[index]
>> +            index = child_index
>> +
>> +    def extract(self): #функция для извлечения
>> +        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
>> +        value=self.__heap.pop() #потом извлекается сюда
>>
>> Происходит удаление последнего элемента из кучи и сохранение его в
>> переменную value
>> ------------------------------
>>
>> In 5 modul/5.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638852776>
>> :
>>
>> > +def main():
>> +    n = int(input())
>> +    do = [0]*4*n
>> +    a = list(map(int, input().split()))[:n]
>> +    build(0, 0, n, do, a)
>> +    q = int(input())
>> +    res = []
>> +    while q != 0:
>> +        que, l, r = map(str, input().split())
>> +        if que=="s":
>> +            res.append(get_gcd(0, 0, n, do, int(l)-1, int(r)))
>> +        else:
>> +            update(0, 0, n, do, int(l)-1, int(r))
>> +        q -= 1
>> +    print(*res)
>> +main()
>>
>> У Вас в 5 модуле на Контесте зачтена только 2 задача, может быть мне
>> поступить также?
>> ------------------------------
>>
>> In 6 modul/1.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638853989>
>> :
>>
>> > +    return child_index+1
>> +
>> +def add(component, heap):
>> +    heap.append(component)
>> +    return shift_up(len(heap)-1, heap)
>> +
>> +
>> +def extract(heap):
>> +    heap[0], heap[len(heap)-1] = heap[len(heap)-1], heap[0]
>> +    id = heap.pop()
>> +    if heap:
>> +        return shift_down(0, heap), id
>> +    else:
>> +        return 0, id
>> +
>> +def min(heap): #поиск минимального
>>
>> Извлечение вершины кучи, построенной по минимальному элементу.
>> ------------------------------
>>
>> In 6 modul/1.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638855902>
>> :
>>
>> > +        if int(data[0]) == 1:
>> +            if not heap:
>> +                result.append(-1)
>> +            else:
>> +                result.append(extract(heap))
>> +        elif int(data[0]) == 2:
>> +            if len(heap) == n:
>> +                result.append(-1)
>> +            else:
>> +                result.append(add(int(data[-1]), heap))
>> +    for i in rasult:
>> +        if type(i) == tuple:
>> +            print(*i)
>> +        else:
>> +            print(i)
>> +    print(*heap)
>>
>> Ну, тут явно нужны комментарии для функции main()
>> Не забываем условие: если копируем код, то он должен быть максимально
>> закомментирован.
>> Это касается всего модуля 6.
>> ------------------------------
>>
>> In 6 modul/3.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638851790>
>> :
>>
>> > +
>> +    def shift_down(self, index):
>> +        while 2*index+1 < self.length:
>> +            left_index = 2*index+1
>> +            right_index = 2*index+2
>> +            child_index = left_index
>> +            if right_index < self.length and self.__heap[left_index] < self.__heap[right_index]:
>> +                child_index = right_index #child_index
>> +            if self.__heap[child_index] <= self.__heap[index]:
>> +                break
>> +            self.__heap[index], self.__heap[child_index] = self.__heap[child_index], self.__heap[index]
>> +            index = child_index
>> +
>> +    def extract(self): #функция для извлечения
>> +        self.__heap[0], self.__heap[self.length-1] = self.__heap[self.length-1], self.__heap[0] #в этой строке проходит извленчения
>> +        value=self.__heap.pop() #потом извлекается сюда
>>
>> Нужно как-то так комментировать.
>> ------------------------------
>>
>> In 6 modul/1.txt
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#discussion_r638856161>
>> :
>>
>> > +        if int(data[0]) == 1:
>> +            if not heap:
>> +                result.append(-1)
>> +            else:
>> +                result.append(extract(heap))
>> +        elif int(data[0]) == 2:
>> +            if len(heap) == n:
>> +                result.append(-1)
>> +            else:
>> +                result.append(add(int(data[-1]), heap))
>> +    for i in rasult:
>> +        if type(i) == tuple:
>> +            print(*i)
>> +        else:
>> +            print(i)
>> +    print(*heap)
>>
>> И кстати, Вы эти задачи даже на Контест не загрузили.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#pullrequestreview-667914222>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ARAEUTOJH3EJFO7KRNJGZJ3TPOZSFANCNFSM43RRTIWA>
>> .
>>
>

Nastasia8 commented on 6 modul/2.txt:
```diff
@@ -0,0 +1,71 @@
+    heap.append(component)
+    return shift_up(len(heap)-1, heap)
+    
+def old_extract(index, heap):
```
Это new)

Nastasia8 commented on 6 modul/2.txt:
```diff
@@ -0,0 +1,71 @@
+                result.append(-1)
+            else:
+                result.append(add(int(data[-1]), heap))
+    for i in rasult:
```
Переименовывать нужно аккуратнее

Nastasia8 commented:
1. "...1 и 2 задача не прошла я их сам делал, .." - т.е. прямо сам?)) https://github.com/Nastasia8/AaDS_1_185_2021/pull/66/commits/1d33d879403ba735000c0a257f9d0240378d7afa
https://github.com/Nastasia8/AaDS_1_185_2021/pull/66/commits/7000047786be56af513508f5376e25984cb76787
Вы допустили ошибку при переименовании.
2. Модуль 5: задача 3 и 5 -> А какая ошибка, какой компилятор?????

Artem22831 commented:
1.в 6 модуле 1 задачу вы мне сами скинули ,т.е я вам скинул , а вы мне
переправили
2. там ошибка при компиляции вот:

вт, 25 мая 2021 г. в 18:59, Nastasia8 ***@***.***>:

>
>    1. "...1 и 2 задача не прошла я их сам делал, .." - т.е. прямо сам?))
>    1d33d87
>    <https://github.com/Nastasia8/AaDS_1_185_2021/commit/1d33d879403ba735000c0a257f9d0240378d7afa>
>    7000047
>    <https://github.com/Nastasia8/AaDS_1_185_2021/commit/7000047786be56af513508f5376e25984cb76787>
>    Вы допустили ошибку при переименовании.
>    2. Модуль 5: задача 3 и 5 -> А какая ошибка, какой компилятор?????
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-847994427>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTLTDYEFXRLBU6M7LFTTPPCNLANCNFSM43RRTIWA>
> .
>

Artem22831 commented:
Сколько у меня баллов за семестр будет?

вт, 25 мая 2021 г. в 19:06, Jumper csgo ***@***.***>:

> 1.в 6 модуле 1 задачу вы мне сами скинули ,т.е я вам скинул , а вы мне
> переправили
> 2. там ошибка при компиляции вот:
>
> вт, 25 мая 2021 г. в 18:59, Nastasia8 ***@***.***>:
>
>>
>>    1. "...1 и 2 задача не прошла я их сам делал, .." - т.е. прямо сам?))
>>    1d33d87
>>    <https://github.com/Nastasia8/AaDS_1_185_2021/commit/1d33d879403ba735000c0a257f9d0240378d7afa>
>>    7000047
>>    <https://github.com/Nastasia8/AaDS_1_185_2021/commit/7000047786be56af513508f5376e25984cb76787>
>>    Вы допустили ошибку при переименовании.
>>    2. Модуль 5: задача 3 и 5 -> А какая ошибка, какой компилятор?????
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-847994427>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ARAEUTLTDYEFXRLBU6M7LFTTPPCNLANCNFSM43RRTIWA>
>> .
>>
>

Nastasia8 commented:
Не вижу ошибки, отправьте скрин экрана, только проверьте, что он отображается.

Artem22831 commented:
[image: image.png]

вт, 25 мая 2021 г. в 19:55, Nastasia8 ***@***.***>:

> Не вижу ошибки, отправьте скрин экрана, только проверьте, что он
> отображается.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848047923>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTOMY3WCINQ5BNPWAD3TPPJBVANCNFSM43RRTIWA>
> .
>

Nastasia8 commented:
Артем, я не знаю, как Вы прикрепляете скрин, но он не отображается. Зайдите https://github.com/Nastasia8/AaDS_1_185_2021/pull/65 и посмотрите, какое я вижу от Вас сообщение.

Artem22831 commented:
Так посмотрите через почту , а не через git.

вт, 25 мая 2021 г. в 20:02, Nastasia8 ***@***.***>:

> Артем, я не знаю, как Вы прикрепляете скрин, но он не отображается.
> Зайдите #65 <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65> и
> посмотрите, какое я вижу от Вас сообщение.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848052410>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTNIKNC5MTL3LANJJZTTPPJ3DANCNFSM43RRTIWA>
> .
>

Nastasia8 commented:
Так, я тоже смотрела через почту.
Вот сообщение:
[image: image.png]

вт, 25 мая 2021 г. в 19:55, Nastasia8 ***@***.***>:

> Не вижу ошибки, отправьте скрин экрана, только проверьте, что он
> отображается.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848047923>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTOMY3WCINQ5BNPWAD3TPPJBVANCNFSM43RRTIWA>
> .
>
В ваших интересах сделать так, чтобы я увидела скрин, а не в моих.

Artem22831 commented:
А так https://prnt.sc/13elh3a

вт, 25 мая 2021 г. в 20:16, Nastasia8 ***@***.***>:

> Так, я тоже смотрела через почту.
> Вот сообщение:
> [image: image.png]
>
> вт, 25 мая 2021 г. в 19:55, Nastasia8 *@*.***>:
>
> Не вижу ошибки, отправьте скрин экрана, только проверьте, что он
> отображается.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> #65 (comment)
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848047923>
> ,
> or unsubscribe
>
> https://github.com/notifications/unsubscribe-auth/ARAEUTOMY3WCINQ5BNPWAD3TPPJBVANCNFSM43RRTIWA
> .
>
> В ваших интересах сделать так, чтобы я увидела скрин, а не в моих.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848061459>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTLXQTHM5BXPVHZXC7TTPPLN5ANCNFSM43RRTIWA>
> .
>

Nastasia8 commented:
Вот теперь вижу, у Вас ошибка с синтаксисом. Посмотрите функцию def main() -> все, что ниже должно входить в нее (т.е. у Вас не хватает отступа) до main()

Artem22831 commented:
Всё равно не работает https://prnt.sc/

вт, 25 мая 2021 г. в 20:29, Nastasia8 ***@***.***>:

> Вот теперь вижу, у Вас ошибка с синтаксисом. Посмотрите функцию def main()
> -> все, что ниже должно входить в нее (т.е. у Вас не хватает отступа) до
> main()
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848072299>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTJQDWD3TW3YIT2SNXDTPPNBLANCNFSM43RRTIWA>
> .
>

Artem22831 commented:
Всё равно не работает https://prnt.sc/13em790

вт, 25 мая 2021 г. в 20:29, Nastasia8 ***@***.***>:

> Вот теперь вижу, у Вас ошибка с синтаксисом. Посмотрите функцию def main()
> -> все, что ниже должно входить в нее (т.е. у Вас не хватает отступа) до
> main()
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848072299>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTJQDWD3TW3YIT2SNXDTPPNBLANCNFSM43RRTIWA>
> .
>

Nastasia8 commented:
У 34 строки тоже должен быть отступ

Artem22831 commented:
Мне пишет в компиляторе,что ошибка в 10 строке invalid syntax

вт, 25 мая 2021 г. в 20:43, Nastasia8 ***@***.***>:

> У 34 строки тоже должен быть отступ
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848081634>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARAEUTP22ODW7IR3RFU6JP3TPPOTFANCNFSM43RRTIWA>
> .
>

Nastasia8 commented:
Потому что функция gcd() должна быть, а не global()

Artem22831 commented:
Теперь ошибка в 1 строке https://prnt.sc/13enfal

вт, 25 мая 2021 г. в 20:47, Jumper csgo ***@***.***>:

> Мне пишет в компиляторе,что ошибка в 10 строке invalid syntax
>
> вт, 25 мая 2021 г. в 20:43, Nastasia8 ***@***.***>:
>
>> У 34 строки тоже должен быть отступ
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848081634>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ARAEUTP22ODW7IR3RFU6JP3TPPOTFANCNFSM43RRTIWA>
>> .
>>
>

Artem22831 commented:
А можно я вам эту задачу просто на гит загружу , на контесте не работает
никак,хотя всё правильно

вт, 25 мая 2021 г. в 21:08, Jumper csgo ***@***.***>:

> Теперь ошибка в 1 строке https://prnt.sc/13enfal
>
> вт, 25 мая 2021 г. в 20:47, Jumper csgo ***@***.***>:
>
>> Мне пишет в компиляторе,что ошибка в 10 строке invalid syntax
>>
>> вт, 25 мая 2021 г. в 20:43, Nastasia8 ***@***.***>:
>>
>>> У 34 строки тоже должен быть отступ
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Nastasia8/AaDS_1_185_2021/pull/65#issuecomment-848081634>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ARAEUTP22ODW7IR3RFU6JP3TPPOTFANCNFSM43RRTIWA>
>>> .
>>>
>>

Artem22831 committed:
Update 3.txt

Artem22831 committed:
Update 1.txt

Artem22831 committed:
Update 3.txt

Artem22831 committed:
Update 5.txt
